### PR TITLE
[Squid][SMB][Automation]Clustering support

### PR DESCRIPTION
**Pull request by MohitBis**
[Squid][SMB][Automation]Clustering support

Scenarios:
1. (Imperative Style - Single Share) Verify Samba deployment using auth_mode=user with clustering
2. (Imperative Style - Multiple Shares) Verify Samba deployment using auth_mode=user with clustering
3. (Imperative Style - Single Share) Verify Samba deployment using auth_mode=AD with clustering
4. (Imperative Style - Multiple Share) Verify Samba deployment using auth_mode=AD with clustering
5. (Declarative Style - Single Share) Verify Samba deployment using auth_mode=user with clustering
6. (Declarative Style - Multiple Share) Verify Samba deployment using auth_mode=user with clustering
7. (Declarative Style - Single Share) Verify Samba deployment using auth_mode=AD with clustering
8. (Declarative Style - Multiple Share) Verify Samba deployment using auth_mode=AD with clustering
9. (Imperative Style) Verify Samba deployment with clustering=never
10. (Declarative Style) Verify Samba deployment with clustering=never

Note:
1. Multi share with clustering currently failing with open bug

### CODE REVIEW

**Comment by MohitBis** [comment]
Logs: http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-4K43TU/

**Review by pranavprakash20 — APPROVED** [review]
LGTM

File: tests/smb/smb_deployment_declarative_method.py [hunk 1]
```diff
@@ -84,6 +94,17 @@ def run(ceph_cluster, **kw):
             file_mount,
```
Comment (adityaramteke) [inline tests/smb/smb_deployment_declarative_method.py:94]: Did we miss adding a clustering parameter for declarative deployment?

File: suites/squid/smb/tier-1-smb-ctdb-clustering.yaml [hunk 2]
```diff
@@ -0,0 +1,347 @@
+tests:
```
[the remark this answers is not in the document]
Comment (MohitBis) [reply suites/squid/smb/tier-1-smb-ctdb-clustering.yaml:7]: Thanks @adityaramteke 

Updated

File: tests/smb/smb_operations.py [hunk 3]
```diff
@@ -591,3 +597,63 @@ def clients_cleanup(
...
+        smb_nodes (list): samba server nodes obj list
+        smb_cluster_id (str): samba cluster id
+    """
+    try:
+        # Get samba service
+        cmd = f"cephadm ls --no-detail | jq -r 'map(select(.name | startswith(\"smb.{smb_cluster_id}\")))[-1].name'"
+        out = smb_nodes[0].exec_command(sudo=True, cmd=cmd)[0].strip()
+
+        # Get ctdb status
+        sleep(30)
```
[the remark this answers is not in the document]
Comment (MohitBis) [reply tests/smb/smb_operations.py:641]: Hi @adityaramteke 

Currently after deployment of samba services with clustering, ctdb status  taking time to reflect the status.
We have open bug for same.

Once it resolved we will remove the sleep

Thanks,
Mohit Bisht

File: tests/smb/smb_operations.py [hunk 4]
```diff
@@ -591,3 +597,63 @@ def clients_cleanup(
...
+        cmd = f"cephadm enter -n {out} ctdb status"
+        ctdb_Status = smb_nodes[0].exec_command(sudo=True, cmd=cmd)[0]
```
[the remark this answers is not in the document]
Comment (MohitBis) [reply tests/smb/smb_operations.py]: Thanks @adityaramteke 
Updated

File: tests/smb/smb_operations.py [hunk 5]
```diff
@@ -591,3 +597,63 @@ def clients_cleanup(
...
+
+        # Get number of nodes and status
+        nodes = re.search(r"Number of nodes:(\d+)", ctdb_Status).group(1)
+        status = all(
+            re.search(r"pnn:\d+ \S+ *OK", line)
+            for line in ctdb_Status.splitlines()
```
[the remark this answers is not in the document]
Comment (MohitBis) [reply tests/smb/smb_operations.py]: Thanks @adityaramteke
Updated

File: tests/smb/smb_deployment_declarative_method.py (hunk 1 above)
Comment (MohitBis) [reply tests/smb/smb_deployment_declarative_method.py:94]:
```suggestion
            file_mount,
```

Comment (MohitBis) [reply tests/smb/smb_deployment_declarative_method.py:94]: Hi @adityaramteke 

No,

Clustering is by default enabled.

Ref: https://docs.ceph.com/en/latest/mgr/smb/

Thanks,
Mohit Bisht

**Review by adityaramteke — APPROVED** [review]
LGTM, nice work!

**Review by pranavprakash20 — APPROVED** [review]
LGTM

**Review by tintumathew10 — APPROVED** [review]
LGTM

**Pull request merged**